apps/mobile/src/components/app-update-required/update-button.tsx

```diff
@@ -0,0 +1,55 @@
+import { useState } from 'react';
+import { Alert } from 'react-native';
+
+import { getStoreButtonText, handleStoreRedirect } from '@/utils/store-utils';
+import { t } from '@lingui/macro';
+
+import { Button } from '@leather.io/ui/native';
+
+interface UpdateButtonProps {
+  onPress?: () => void;
+}
+
+export function UpdateButton({ onPress }: UpdateButtonProps) {
+  const [isLoading, setIsLoading] = useState(false);
+
+  async function handlePress() {
+    if (onPress) {
+      onPress();
+      return;
+    }
+
+    setIsLoading(true);
+
+    try {
+      await handleStoreRedirect();
+    } catch (error) {
+      // Show user-friendly error message
+      Alert.alert(
+        t({ id: 'version_guard.store_error_title', message: 'Unable to Open Store' }),
+        error instanceof Error
```

Comment:
We have [isError](https://github.com/leather-io/mono/blob/9c1837d596269dfae743372bca8c174f04d8c230/packages/utils/src/index.ts#L57) in `utils`